notebooks/07 Model comparison.ipynb

```diff
@@ -0,0 +1,492 @@
+    "    for proportion in risks:\n",
+    "        if proportion == \"1 - Very Low Risk\":\n",
+    "            data = risks[proportion]\n",
+    "            axs[i, 0].bar(risk_labels, data, label=proportion, width=0.35)\n",
+    "        else:\n",
+    "            bottom += data\n",
+    "            data = risks[proportion]\n",
+    "            axs[i, 0].bar(\n",
+    "                risk_labels, data, label=proportion, bottom=bottom, width=0.35\n",
+    "            )\n",
+    "    # skip legend as will be same on RHS\n",
+    "    # handles, labels = axs[i, 0].get_legend_handles_labels()\n",
+    "    # axs[i, 0].legend(handles[::-1], labels[::-1], bbox_to_anchor=(1.05, 1))\n",
+    "    axs[i, 0].set_xlabel(\"Actual risk\")\n",
+    "    axs[i, 0].set_ylabel(\"Predicted risk\")\n",
+    "    axs[i, 0].set_title(f\"classification: {model_classification}\")\n",
+    "\n",
+    "    # plot actual vs predicted CLASSES for regression\n",
+    "    risks = dict.fromkeys(risk_labels)\n",
+    "    for proportion in risks:\n",
+    "        risks[proportion] = np.array([0.0, 0.0, 0.0, 0.0, 0.0])\n",
+    "\n",
+    "        for label in risk_labels:\n",
+    "            this_risk = int(label[0])\n",
+    "\n",
+    "            # extract the real risk\n",
+    "            subset = predictions_df[predictions_df.risk == this_risk]\n",
+    "\n",
+    "            if proportion == \"1 - Very Low Risk\":\n",
+    "                prop = (subset.pred_regression == 1).sum() / subset.shape[0]\n",
+    "            elif proportion == \"2 - Low Risk\":\n",
+    "                prop = (subset.pred_regression == 2).sum() / subset.shape[0]\n",
+    "            elif proportion == \"3 - Normal Risk\":\n",
+    "                prop = (subset.pred_regression == 3).sum() / subset.shape[0]\n",
+    "            elif proportion == \"4 - Elevated Risk\":\n",
+    "                prop = (subset.pred_regression == 4).sum() / subset.shape[0]\n",
+    "            else:\n",
+    "                prop = (subset.pred_regression == 5).sum() / subset.shape[0]\n",
+    "\n",
+    "            risks[proportion][this_risk - 1] = prop\n",
+    "\n",
+    "    bottom = np.array([0.0, 0.0, 0.0, 0.0, 0.0])\n",
+    "    for proportion in risks:\n",
+    "        if proportion == \"1 - Very Low Risk\":\n",
+    "            data = risks[proportion]\n",
+    "            axs[i, 1].bar(risk_labels, data, label=proportion, width=0.35)\n",
+    "        else:\n",
+    "            bottom += data\n",
+    "            data = risks[proportion]\n",
+    "            axs[i, 1].bar(\n",
+    "                risk_labels, data, label=proportion, bottom=bottom, width=0.35\n",
+    "            )\n",
+    "    handles, labels = axs[i, 1].get_legend_handles_labels()\n",
+    "    axs[i, 1].legend(handles[::-1], labels[::-1], bbox_to_anchor=(1.05, 1))\n",
+    "    axs[i, 1].set_xlabel(\"Actual risk\")\n",
+    "    axs[i, 1].set_ylabel(\"Predicted risk\")\n",
```

Comment:
Should this be Predicted risk (proportion) or something like that?

Reply:
Updated